EIPS/eip-erps/eip-erps.md

```diff
@@ -0,0 +1,194 @@
+A JSON object that includes a “type” field and a set of “data” fields, where the contents of the data fields is dependent on the type and must follow the schema define for that type.  This allows clients to support a limited set of operations initially and add more based on a subsequent EIP if needed.  Support for the following action types should be implemented by clients upon adoption of this EIP:  
+
+#### weiTransfer
+Transfers eth from one address to another.
+
+*Data fields*
+
+- **type** (*string*): `weiTransfer`
+- **fromAddress** (*hex string*): The address from which each should be transferred
+- **toAddress** (*hex string*): The address to which eth should be sent
+- **value** (*decimal string*): The amount to be transferred, in units of wei. The value must be a whole number and non-negative.
+
+
+#### storeCode
+Stores the given code at the given address.
+
+*Data fields*
+
+- **type** (*string*): `storeCode`
+- **toAddress** (*hex string*): The address to which the contract should be restored.  
+- **expectedCodeHash** (*hex string*): the expected hash of the code already at the toAddress.The empty string should be used if no code is expected at the toAddress.  In all other cases, the “0x” prefix is optional.
+- **code** (*hex string*): the new byte code associated with the contract
+
+### Appendix (Optional)
+The appendix can include additional supporting evidence or attachments that will help reviewers understand or verify the claims made in the ERP.  
+
+For the storeCode operation, it should include the proposed contract source (e.g. solidity) as well as the other details required such that a reviewer can compile the source and generate the same byte code.  It should also include the source that was originally stored at that address, if possible/applicable.  If the two contracts are not identical, changes should be noted. If they are identical, the author should indicate why no changes are necessary (this is unlikely). Additionally, any relevant reviews, audits, and test cases should be included to the extent that they address the issues encountered with the original contract.  
+
+If accepted, an ERP could easily compile the block at while the changes are to take place, and the source of the actions (which would be the output of the script, in a standardized object output). These can be bundled with the client for seamless execution.
+
+
+## ERP Approval Process
+Approval of any ERP will require a mandatory minimum of 30-days between the time the ERP is submitted (or substantially modified) and adoption by clients, and ERPs affecting more funds or more addresses may prompt a considerably longer review period.  In practice, ERPs will often require more time because the will likely be bundled into other client upgrades that are on their own schedule.  However, the 30-day mandatory review period  guarantees a period for comment and auditing and prevents any last minute inclusions from being rushed into an upcoming upgrade.  
```

Comment:
[comment not as an EIP editor] I would prefer another mandatory period between the acceptance of ERP and the enforcement of the ERP.  This gives chance for people to fork away from Ethereum if they disagree.

Reply:
I don't think a separate period is necessary.  The purpose of the ERP is to determine if the proof of ownership is clear, supported by convincing evidence.  If some claim in the ERP is not well supported, people are free to comment, provide counter-evidence, or suggest what sort of proof might be more convincing.  

However, if the ERP *does* provide strong evidence, no reasonable argument can be made counter to the original claim, and the affected parties are also in agreement about the outcome, I think providing another period where people can opt-out because they disagree with the *facts that have been demonstrated to be true* is counterproductive.